src/game/SaveLoadGame.cc

```diff
@@ -1132,6 +1138,85 @@ static void SaveMercProfiles(HWFILE const f)
 	}
 }
 
+const char * IMPSavedProfileCreateFilename(const char *nickname_cstring)
+{
+	char *profile_filename = (char *)malloc(strlen(GCM->getSavedGamesFolder().c_str()) + 32);
+	sprintf(profile_filename, "%s/mercprofile.%s",GCM->getSavedGamesFolder().c_str(), nickname_cstring);
+	return profile_filename;
+}
+
+bool IMPSavedProfileDoesFileExist(const char *nickname)
+{
+	const char *profile_filename = IMPSavedProfileCreateFilename(nickname);
+	bool fexists = Fs_exists(profile_filename);
+	free((void*)profile_filename);
+	return fexists;
+}
+
+SGPFile* const IMPSavedProfileOpenFileForRead(const char *nickname)
+{
+	if (!IMPSavedProfileDoesFileExist(nickname)) return (SGPFile *)-1;
+	const char *profile_filename = IMPSavedProfileCreateFilename(nickname);
+	SGPFile *f = FileMan::openForReading(profile_filename);
+	free((void*)profile_filename);
+	return f;
+}
+
+SGPFile* const IMPSavedProfileOpenFileForWrite(const char *nickname)
+{
+	const char *profile_filename = IMPSavedProfileCreateFilename(nickname);
+	SGPFile *f = FileMan::openForWriting(profile_filename, true);
+	free((void*)profile_filename);
+	return f;
+}
+
+int IMPSavedProfileLoadMercProfile(const char *nickname)
+{
+	if (!IMPSavedProfileDoesFileExist(nickname)) return -1;
+	SGPFile *f = IMPSavedProfileOpenFileForRead(nickname);
+	MERCPROFILESTRUCT profile_saved;
+	FileRead(f, &profile_saved, sizeof(MERCPROFILESTRUCT));
+	FileClose(f);
+	int voiceid = profile_saved.ubSuspiciousDeath;
+	MERCPROFILESTRUCT& profile_new = gMercProfiles[PLAYER_GENERATED_CHARACTER_ID + voiceid];
+	profile_new = profile_saved;
+	profile_new.bMercStatus = MERC_OK;
+	return voiceid;
+}
+
+void IMPSavedProfileLoadInventory(const char *nickname, SOLDIERTYPE *pSoldier)
+{
+	if (!IMPSavedProfileDoesFileExist(nickname)) return;
+	if (!pSoldier) return;
+
+	SGPFile *f = IMPSavedProfileOpenFileForRead(nickname);
+	FileSeek(f, sizeof(MERCPROFILESTRUCT), FILE_SEEK_FROM_START);
+	FileRead(f, pSoldier->inv, sizeof(OBJECTTYPE) * NUM_INV_SLOTS);
+	FileClose(f);
+}
+
+void SaveIMPPlayerProfiles()
+{
+	#define LASTIMPPROFILE NPC169 // (PLAYER_GENERATED_CHARACTER_ID+5)
+
+	for (int i = PLAYER_GENERATED_CHARACTER_ID; i <= LASTIMPPROFILE; i++)
+	{
+		MERCPROFILESTRUCT* const mercprofile = &gMercProfiles[i];
+		if (mercprofile->bLife == 0) continue;
+		SOLDIERTYPE* const pSoldier = FindSoldierByProfileID(i);
+		if (!pSoldier) continue;
+		if (pSoldier->bTeam != OUR_TEAM) continue;
+		if (pSoldier->ubWhatKindOfMercAmI != MERC_TYPE__PLAYER_CHARACTER) continue;
+
+		SGPFile *f = IMPSavedProfileOpenFileForWrite(mercprofile->zNickname.c_str());
+		if (!f) continue;
+
+		mercprofile->ubSuspiciousDeath = i - PLAYER_GENERATED_CHARACTER_ID; // save voice_id, field not used for resuscitated merc
```

Comment:
Rather than doing this hack, why not just write the Profile ID and Portrait ID into the file along with the structs?

This is also what JA 1.13 does : https://github.com/aimnas/ja2-1.13-source-mirror/blob/b988df876f36e373ca21e0001a91aefe87ecdd50/Laptop/IMP%20Confirm.cpp#L1456

Reply:
The ideal would be to use JSON to export/import to abstract from the compiler-struct memory format. This is not difficult. A text file would make it easier to edit for creating mod scenarios such as the intended "All you need is kill".

The data was placed in an unrelated field because it was an easy hack.